kmp-nativecoroutines-compiler/src/main/kotlin/com/rickclephas/kmp/nativecoroutines/compiler/KmpNativeCoroutinesIrTransformer.kt

```diff
@@ -178,4 +188,31 @@ internal class KmpNativeCoroutinesIrTransformer(
             dispatchReceiver = function.dispatchReceiverParameter?.let { irGet(it) }
         }
     }
+
+    private val propagatedExceptionClasses = propagatedExceptions.map {
+        context.referenceClass(it) ?: throw NoSuchElementException("Couldn't find $it symbol")
+    }
+    private val propagatedExceptionsArrayElementType = context.irBuiltIns.kClassClass.typeWithArguments(listOf(
+        makeTypeProjection(context.irBuiltIns.throwableType, Variance.OUT_VARIANCE)
+    ))
+
+    private fun IrBuilderWithScope.createPropagatedExceptionsArray(originalDeclaration: IrDeclaration): IrExpression {
+        // Find the annotation on the declaration (or a parent class)
+        fun IrDeclaration.getAnnotation(): IrConstructorCall? =
+            annotations.findNativeCoroutineThrowsAnnotation() ?:
+            useThrowsAnnotation.ifTrue { originalDeclaration.annotations.findThrowsAnnotation() } ?:
+            parentClassOrNull?.getAnnotation()
```

Comment:
Maybe this should just use all the annotations, would probably make more sense since the `propagatedExceptions` are also applied (even if the function has an annotation).